src/utils/weather_model.jl

```diff
@@ -0,0 +1,215 @@
+
+"""
+    weather_model_data_path(start_date, start_time)
+
+Get the path to the weather model data for a given start date and time.
+If the data is not found, will attempt to generate it from raw data. If 
+the raw data is not found, throw an error
+
+# Arguments
+- `start_date`: Start date as string YYYYMMDD
+- `start_time`: Start time as string HHMM
+"""
+function weather_model_data_path(start_date, start_time)
+    # get data path
+    ic_data_path = joinpath(
+        @clima_artifact("weather_model_ic"),
+        "init",
+        "era5_init_$(start_date)_$(start_time).nc",
+    )
+    raw_data_path = joinpath(
+        @clima_artifact("weather_model_ic"),
+        "raw",
+        "era5_raw_$(start_date)_$(start_time).nc",
+    )
+
+    if !isfile(ic_data_path)
+        @info "Initial condition file $ic_data_path does not exist. Attempting to generate it now..."
+        if !isfile(raw_data_path)
+            day = Dates.format(Date(start_date, "yyyymmdd"), "yyyy-mm-dd")
+            time = Dates.format(Time(start_time, "HHMM"), "HH:MM")
+            error(
+                "Source file $(raw_data_path) does not exist. Please run `python get_initial_conditions.py --output-dir $(@clima_artifact("weather_model_ic"))/raw --date $(day) --time $(time)` to download the data.",
+            )
+        end
+        # get target levels - TODO: make this more flexible
+        target_level_path =
+            joinpath(@clima_artifact("weather_model_ic"), "target_levels.txt")
+        target_levels = parse.(Float32, readlines(target_level_path))
+
+        @info "Interpolating raw weather model data onto z-levels"
+        to_z_levels(raw_data_path, ic_data_path, target_levels, Float32)
+    end
+    return ic_data_path
+end
+
+
+"""
+    to_z_levels(source_file, target_file, target_levels, FT)
+
+Interpolate ERA5 data from native model levels to specified z-levels. Note that 
+to match MoistFromFile, we rename surface pressure, sp, to p. This should change 
+for consistency if we write a new initial condition dispatch.
+
+# Arguments
+- `source_file` / `target_file`: Path to input / output NetCDF 
+- `target_levels`: Vector of target altitude levels (in meters)
+
+"""
+function to_z_levels(source_file, target_file, target_levels, FT)
+
+    param_set = TD.Parameters.ThermodynamicsParameters(FT)
+    grav = TD.Parameters.grav(param_set)
+
+    ncin = Dataset(source_file)
+
+    # Read and cast coordinates to FT type
+    lat = FT.(ncin["latitude"][:])
+    lon = FT.(ncin["longitude"][:])
+
+    # Read and cast variables to FT type, replacing missing values with NaN
+    z_raw = ncin["z"][:, :, :, 1]
+    source_z = FT.(coalesce.(z_raw, NaN)) ./ grav # convert from geopotential height to height
+
+    # Create output file
+    ncout = NCDataset(target_file, "c", attrib = copy(ncin.attrib))
+
+    # Define dimensions
+    defDim(ncout, "lon", size(lon, 1))
+    defDim(ncout, "lat", size(lat, 1))
```

Comment:
You should be able to use `length`.

```suggestion
    defDim(ncout, "lon", length(lon))
    defDim(ncout, "lat", length(lat))
```